rust/gateway/src/eventloop.rs

```diff
@@ -65,10 +65,9 @@ impl Eventloop {
 
                     continue;
                 }
-                Poll::Ready(Event::ConnectionIntent { .. }) => {
-                    unreachable!("Not used on the gateway, split the events!")
+                Poll::Ready(Err(e)) => {
+                    tracing::warn!("Tunnel error: {e}");
```

Comment:
Similarly to the client, the gateway now doesn't fail on tunnel errors. It actually did before which is probably not good because that interrupts all connections.

Similarly to the clients, we need to think about which errors should actually shut down the gateway.